File: dart/ductus/lib/src/adapter/candidates.dart
/// Shared intermediate structures of the manual input paths (A: comments,
/// B: Dart annotations).
library;

import 'graph_model.dart';

/// A `@journey:action` / `@JourneyAction` whose `from` may only be resolved
/// after scanning all files, via the enclosing class.
class ActionCandidate {
  final String? id;
  final String label;
  final String to;
  final String? from;
  final String trigger;
  final String? condition;

  /// Name of the enclosing class (for from inference), if any.
  final String? enclosingClassName;
  final SourceRef sourceRef;

  const ActionCandidate({
    this.id,
    required this.label,
    required this.to,
    this.from,
    required this.trigger,
    this.condition,
    this.enclosingClassName,
    required this.sourceRef,
  });
}

/// Result of one manual input path for a file.
class ManualExtraction {
  final List<GraphNode> nodes = [];
  final List<GraphFlow> flows = [];
  final List<ActionCandidate> actions = [];

  /// Class name -> screen id (for from inference and context.go mapping).
  final Map<String, String> screenClassNames = {};
}

File: packages/adapter-typescript/src/candidates.ts
/**
 * Intermediate structures of the manual extraction (path A) whose `from`
 * may only be resolved after ALL files have been scanned.
 */

import type { GraphFlow, GraphNode, SourceRef } from './graph-model.js';

/** A @journey:action whose `from` may still need to be resolved. */
export interface ActionCandidate {
  id?: string;
  label: string;
  to: string;
  from?: string;
  trigger: string;
  condition?: string;
  /** Name of the enclosing component for `from` inference. */
  enclosingName?: string;
  sourceRef: SourceRef;
}

/** Result of the manual extraction of a single file. */
export class ManualExtraction {
  readonly nodes: GraphNode[] = [];
  readonly flows: GraphFlow[] = [];
  readonly actions: ActionCandidate[] = [];
  /** Component name → screen id (for `from` inference and nav mapping). */
  readonly screenSymbols = new Map<string, string>();
}
